src/meta/src/hummock/compaction/selector/level_selector.rs

```diff
@@ -239,53 +239,62 @@ impl DynamicLevelSelectorCore {
             // range at each level, so the number of levels is the most important factor affecting
             // the read performance. At the same time, the size factor is also added to the score
             // calculation rule to avoid unbalanced compact task due to large size.
-            let non_overlapping_score = {
-                let total_size = levels.l0.as_ref().unwrap().total_file_size
-                    - handlers[0].get_pending_output_file_size(ctx.base_level as u32);
-                let base_level_size = levels.get_level(ctx.base_level).total_file_size;
-                let base_level_sst_count =
-                    levels.get_level(ctx.base_level).table_infos.len() as u64;
-
-                // size limit
-                let non_overlapping_size_score = total_size * SCORE_BASE
-                    / std::cmp::max(self.config.max_bytes_for_level_base, base_level_size);
-                // level count limit
-                let non_overlapping_level_count = levels
-                    .l0
-                    .as_ref()
-                    .unwrap()
-                    .sub_levels
-                    .iter()
-                    .filter(|level| level.level_type() == LevelType::Nonoverlapping)
-                    .count() as u64;
-                let non_overlapping_level_score = non_overlapping_level_count * SCORE_BASE
-                    / std::cmp::max(
-                        base_level_sst_count / 16,
-                        self.config.level0_sub_level_compact_level_count as u64,
-                    );
-
-                std::cmp::max(non_overlapping_size_score, non_overlapping_level_score)
-            };
+            let total_size = levels
+                .l0
+                .as_ref()
+                .unwrap()
+                .sub_levels
+                .iter()
+                .filter(|level| {
```

Comment:
question: if we filter the sub-level of non-partition, will it affect the Intra trigger? IIUC, we are relying on Intra's `pick_whole_level` to merge and align smaller sub-levels, is this not a contradiction?